### DISCUSSION TITLE
how to raise intent to open an web app in a new different url based on context value

### DISCUSSION THREAD

**Discussion by BreezySYP** [finos/FDC3#1047] [Q&A]
I have fdc3 apps, one is a car web app, I have app definition with url http://car.service.com in appd, and if I want to visit different car detail I need to enter http://car.service.com/ford/edge

another app is a desktop app AssetAnalysis, how can I achieve that ---- I can open http://car.service.com/ford/edge in my AssetAnalysis without writing any code in my car web app? i.e. how to set up raise intent for AssetAnalysis in appd, and how to set up listen intent for car web app?

below is my image on how to achieve that

```
// in AssetAnalysis appd
...
{
  "raises": 
  {
    "fdc3.mydomain.ViewCars":["/ford/edge"]
  }
}
...


// in car web appd
...
{
  "listensFor":
  {
    "fdc3.mydomain.ViewCars":
    {
      "displayName":"Test Intent",
      "contexts":["fdc3.mydomain.ViewCars"]
    }
  }
}
...

**Comment by kriswest** [comment]
Hi @BreezySYP, how an app responds to a raised intent has to be controlled by that application, by design. If you want the car web app to navigate to a particular URL on receiving an intent it needs to do that itself in the intent handler.  I.e. the application launches with its default URL, adds an intent listener (fdc3.addIntentListener) to receive the intent and context, receives it and then navigates to the desired URL. Ideally, it should work out this URL itself based on the input rather than receiving a URL directly in the context (as this might open it up to be navigated to pages it should not be).

This is because the FDC3 API is designed to allow apps to opt-in to interop with other applications, rather than allowing them to be controlled by other apps. Hence, all handling behavior, beyond the launch of the application via its default URL, is implemented by the apps themselves.
